fix(bitgo): getUnspentInfo to handle missing unspents

DariusParvin opened the pull request:
Ticket: BG-48421

DariusParvin commented on modules/bitgo/src/v2/coins/utxo/recovery/crossChainRecovery.ts:
```diff
@@ -47,8 +46,24 @@ class BitgoPublicApi {
   async getUnspentInfo(addresses: string[]): Promise<Unspent[]> {
-    const url = this.coin.url(`/public/addressUnspents/${_.uniq(addresses).join(',')}`);
-    return (await request.get(url)).body as Unspent[];
+    const res = await Bluebird.map(
```
I just realized that at this point it would make sense to get rid of any duplicate addresses in `addresses`. I'm pushing a commit for it.

mmcshinsky-bitgo commented on modules/bitgo/src/v2/coins/utxo/recovery/crossChainRecovery.ts:
```diff
@@ -47,8 +46,25 @@ class BitgoPublicApi {
   async getUnspentInfo(addresses: string[]): Promise<Unspent[]> {
-    const url = this.coin.url(`/public/addressUnspents/${_.uniq(addresses).join(',')}`);
-    return (await request.get(url)).body as Unspent[];
+    const uniqueAddresses = new Set(addresses);
```
Is there any merit to doing the one call first to the API and then attempting this secondary approach only if it is 404?

DariusParvin replied:
yeah that's a good idea, i'll add that now. Thanks!

DariusParvin commented:
@mmcshinsky-bitgo ready for re-review. Sorry the force push on [a3cc42a](https://github.com/BitGo/BitGoJS/commit/a3cc42a4305ca55b83500739c3d0b499b23a7a21) was premature. I added the change you suggested and updated the tests (with some minor improvements). Here is the diff since your previous review:

https://github.com/BitGo/BitGoJS/compare/cef988f..8fe1ae9094c728f25f0cc3a53ec170c1c348b49d
